src/PIL/DdsImagePlugin.py

```diff
@@ -124,114 +335,162 @@ def _open(self):
         if len(header_bytes) != 120:
             msg = f"Incomplete header: {len(header_bytes)} bytes"
             raise OSError(msg)
-        header = BytesIO(header_bytes)
+        header = io.BytesIO(header_bytes)
 
         flags, height, width = struct.unpack("<3I", header.read(12))
         self._size = (width, height)
-        self._mode = "RGBA"
 
         pitch, depth, mipmaps = struct.unpack("<3I", header.read(12))
         struct.unpack("<11I", header.read(44))  # reserved
 
         # pixel format
-        pfsize, pfflags = struct.unpack("<2I", header.read(8))
-        fourcc = header.read(4)
-        (bitcount,) = struct.unpack("<I", header.read(4))
+        pfsize, pfflags, fourcc, bitcount = struct.unpack("<4I", header.read(16))
         masks = struct.unpack("<4I", header.read(16))
-        if pfflags & DDPF_LUMINANCE:
-            # Texture contains uncompressed L or LA data
-            if pfflags & DDPF_ALPHAPIXELS:
-                self._mode = "LA"
-            else:
-                self._mode = "L"
-
-            self.tile = [("raw", (0, 0) + self.size, 0, (self.mode, 0, 1))]
-        elif pfflags & DDPF_RGB:
+        if flags & DDSD.CAPS:
+            header.seek(20, io.SEEK_CUR)
```

Comment:
https://learn.microsoft.com/en-us/windows/win32/direct3ddds/dds-header#members states that `DDSD_CAPS` is "Required in every .dds file."

So why do you think this will appear in some files but not others?

Reply:
Just a safety check. What if we get weird dds that have no caps flag and no caps section?

Reply:
The website continues with:
> When you write .dds files, you should set the DDSD_CAPS and DDSD_PIXELFORMAT flags, and for mipmapped textures you should also set the DDSD_MIPMAPCOUNT flag. However, when you read a .dds file, you should not rely on the DDSD_CAPS, DDSD_PIXELFORMAT, and DDSD_MIPMAPCOUNT flags being set because some writers of such a file might not set these flags.

It's not clear to me whether this means readers should always act as if the flag is set, or if a file can be readable without this data. Either way, if the flag is missing, it just means the caps section is set to unknown data, not that it is missing (that would require the `header_size` value to be less than 124). I don't see anywhere the flags value or caps section is actually used when reading a file in Pillow, so I don't see what purpose the seek call has?

Reply:
I've added a commit to https://github.com/REDxEYE/Pillow/pull/7 to remove the seek call.